### ISSUE TITLE
Cannot build project?

### ISSUE THREAD

**Issue by rygo6** [bchavez/RethinkDb.Driver#133]
When I run build.cmd it fails to complete. Here is the terminal output:

```
Paket version 4.4.0
Skipping resolver for group Main since it is already up-to-date
C:\Users\Ryan\Developer\RethinkDb.Driver\Source\paket.lock is already up-to-date
0 seconds - ready.
Paket version 4.4.0
Resolving packages for group build:
 - FAKE 4.61.2
 - NuGet.CommandLine 3.5.0
 - DotNetZip 1.10.1
 - FSharp.Data 2.4.6
 - Newtonsoft.Json 11.0.2
 - secure-file 1.0.31
 - Z.ExtensionMethods.WithNamespace 2.0.13
 - Microsoft.CSharp 4.4.1
 - System.ComponentModel.TypeConverter 4.3.0
 - System.Runtime.Serialization.Primitives 4.3.0
 - System.Runtime.Serialization.Formatters 4.3.0
 - System.Xml.XmlDocument 4.3.0
 - FSharp.Core 4.3.4
 - Zlib.Portable 1.11.0
 - NETStandard.Library 2.0.3
 - System.Reflection.TypeExtensions 4.4.0
 - System.Net.Http 4.3.3
 - System.Collections 4.3.0
 - System.ComponentModel 4.3.0
 - System.ComponentModel.Primitives 4.3.0
 - System.Globalization 4.3.0
 - System.Reflection 4.3.0
 - System.Reflection.Extensions 4.3.0
 - System.Reflection.Primitives 4.3.0
 - System.Resources.ResourceManager 4.3.0
 - System.Runtime 4.3.0
 - System.Runtime.Extensions 4.3.0
 - System.Threading 4.3.0
 - System.Dynamic.Runtime 4.3.0
 - System.Diagnostics.Debug 4.3.0
 - System.Diagnostics.Tools 4.3.0
 - System.IO 4.3.0
 - System.Linq 4.3.0
 - System.Linq.Expressions 4.3.0
 - System.Net.Primitives 4.3.0
 - System.ObjectModel 4.3.0
 - System.Text.Encoding 4.3.0
 - System.Text.Encoding.Extensions 4.3.0
 - System.Text.RegularExpressions 4.3.0
 - System.Threading.Tasks 4.3.0
 - System.Xml.ReaderWriter 4.3.1
 - System.Xml.XDocument 4.3.0
 - Microsoft.Win32.Primitives 4.3.0
 - System.AppContext 4.3.0
 - System.Collections.Concurrent 4.3.0
 - System.Console 4.3.1
 - System.Diagnostics.Tracing 4.3.0
 - System.Globalization.Calendars 4.3.0
 - System.IO.Compression 4.3.0
 - System.IO.Compression.ZipFile 4.3.0
 - System.IO.FileSystem 4.3.0
 - System.IO.FileSystem.Primitives 4.3.0
 - System.Net.Sockets 4.3.0
 - System.Runtime.Handles 4.3.0
 - System.Runtime.InteropServices 4.3.0
 - System.Runtime.InteropServices.RuntimeInformation 4.3.0
 - System.Runtime.Numerics 4.3.0
 - System.Security.Cryptography.Algorithms 4.3.1
 - System.Security.Cryptography.Encoding 4.3.0
 - System.Security.Cryptography.Primitives 4.3.0
 - System.Security.Cryptography.X509Certificates 4.3.2
 - System.Threading.Timer 4.3.0
 - Microsoft.NETCore.Platforms 2.0.2
Paket failed with:
        Could not find versions for package System.Reflection.Emit on https://www.nuget.org/api/v2.
```

Any ideas?

**Comment by bchavez** [comment]
Hi Ryan,

I appologize for that. Pull the latest `master` and try building again. Something happened in **NuGet** or **paket** since the last successful build. Not sure what, but `master` @ 90b2af5 should have the fixes necessary to get past the `System.Reflection.Emit` package error.

Thanks,
Brian

:briefcase: :necktie: ***["Taking care of business every day... Taking care of business every way..."](https://www.youtube.com/watch?v=mmwic9kFx2c)***

**Comment by rygo6** [comment]
Thank you. That got me past that point. Unfortunately a new issue came up:

```
---------------------------------------------------------------------
Build Time Report
---------------------------------------------------------------------
Target      Duration
------      --------
Clean       00:00:00.1307250
restore     00:00:02.5857709
astgen      00:00:03.7972671
BuildInfo   00:00:00.0979321
msb         Failure
Total:      00:00:10.1076479
---------------------------------------------------------------------
Status:     Failure
---------------------------------------------------------------------
---------------------------------------------------------------------
  1) Building C:\Users\Ryan\Developer\RethinkDb.Driver\Source\RethinkDb.Driver\RethinkDb.Driver.csproj failed with exitcode 1.
  2) CS1003: Net\Connection.cs(221,85): Syntax error, '(' expected
  3) CS1031: Net\Connection.cs(221,85): Type expected
  4) CS1026: Net\Connection.cs(221,86): ) expected
  5) CS1003: Net\Connection.cs(230,94): Syntax error, '(' expected
  6) CS1031: Net\Connection.cs(230,94): Type expected
  7) CS1026: Net\Connection.cs(230,95): ) expected
  8) CS1003: Net\Cursor.cs(142,86): Syntax error, '(' expected
  9) CS1031: Net\Cursor.cs(142,86): Type expected
 10) CS1026: Net\Cursor.cs(142,87): ) expected
 11) CS1003: Ast\ReqlAst.cs(69,130): Syntax error, '(' expected
 12) CS1031: Ast\ReqlAst.cs(69,130): Type expected
 13) CS1026: Ast\ReqlAst.cs(69,131): ) expected
 14) CS1003: Ast\ReqlAst.cs(99,127): Syntax error, '(' expected
 15) CS1031: Ast\ReqlAst.cs(99,127): Type expected
 16) CS1026: Ast\ReqlAst.cs(99,128): ) expected
 17) CS1003: Ast\ReqlAst.cs(168,138): Syntax error, '(' expected
 18) CS1031: Ast\ReqlAst.cs(168,138): Type expected
 19) CS1026: Ast\ReqlAst.cs(168,139): ) expected
 20) CS1003: Ast\ReqlAst.cs(209,128): Syntax error, '(' expected
 21) CS1031: Ast\ReqlAst.cs(209,128): Type expected
 22) CS1026: Ast\ReqlAst.cs(209,129): ) expected
 23) CS1003: Ast\ReqlAst.cs(249,130): Syntax error, '(' expected
 24) CS1031: Ast\ReqlAst.cs(249,130): Type expected
 25) CS1026: Ast\ReqlAst.cs(249,131): ) expected
 26) CS1003: Ast\ReqlAst.cs(295,132): Syntax error, '(' expected
 27) CS1031: Ast\ReqlAst.cs(295,132): Type expected
 28) CS1026: Ast\ReqlAst.cs(295,133): ) expected
 29) CS1003: Ast\ReqlAst.cs(330,147): Syntax error, '(' expected
 30) CS1031: Ast\ReqlAst.cs(330,147): Type expected
 31) CS1026: Ast\ReqlAst.cs(330,148): ) expected
 32) CS1003: Ast\ReqlAst.cs(371,52): Syntax error, '(' expected
 33) CS1031: Ast\ReqlAst.cs(371,52): Type expected
 34) CS1026: Ast\ReqlAst.cs(371,53): ) expected
---------------------------------------------------------------------
```

I was also getting this issue when I was trying to build it through the Visual Studio Solution.

**Comment by bchavez** [comment]
Hey Ryan,

What version of **Visual Studio 2017** do you have installed? The `RethinkDb.Driver` requires C# 7.1 or higher. IIRC, C# 7.1 is only available in **Visual Studio 2017 15.3** or higher.

Also, you'll need the **.NET Core SDK 2.0** or higher for **.NET Core builds**. 

The first line the compiler is choking on is a C# 7.1 feature for `default` literals:
https://github.com/bchavez/RethinkDb.Driver/blob/90b2af55fe1194f5b281a6773d7ac8ae752c03a9/Source/RethinkDb.Driver/Net/Connection.cs#L230

My **Visual Studio 2017** version is 15.6.6:

![devenv_1628](https://user-images.githubusercontent.com/478118/40252639-cf151986-5a91-11e8-91cd-7b425cf68efc.png)

My **.NET Core SDK** version info:
```
R:\>
λ dotnet --info
.NET Command Line Tools (2.1.200)

Product Information:
 Version:            2.1.200
 Commit SHA-1 hash:  2edba8d7f1

Runtime Environment:
 OS Name:     Windows
 OS Version:  6.1.7601
 OS Platform: Windows
 RID:         win7-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.200\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.7
  Build    : 2d61d0b043915bc948ebf98836fefe9ba942be11
```
----
* You'll want to make sure you're running an updated version of **Visual Studio 2017** for the latest C# compiler features.

* You'll want to make sure soon as you do a fresh checkout, one of the first things you execute is `build restore` before any build ***and*** before opening the SLN file as outlined here: https://github.com/bchavez/RethinkDb.Driver/wiki/Getting-Started

* After `build restore` is done, you can open up the `SLN` solution file and start hacking away. Once `build restore` command is complete, you should be able to build inside **Visual Studio** or from the **command line** whichever you prefer.

Let me know if this helps or if you need more help!

Thanks,
Brian

:chocolate_bar: :cookie: :lollipop: [**Ronald Jenkees - Stay Crunchy**](https://www.youtube.com/watch?v=GeIAXlwVlZc)

**Comment by bchavez** [comment]
Also, here is the CI build log after we fixed 90b2af5 in `master`.

https://ci.appveyor.com/project/bchavez/rethinkdb-driver/branch/master?fullLog=true

Might be helpful to follow and compare the successful CI build log to your build log to see where things might be going wrong.

:crown: :gem: ***["I know everything that shine ain't always gold..."](https://www.youtube.com/watch?v=7xzU9Qqdqww)***
